resteasy-cdi/src/main/resources/META-INF/beans.xml

```diff
@@ -1,6 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<beans xmlns="http://java.sun.com/xml/ns/javaee" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```

Comment:
Why do you remove this?
Asking because according to [CDI specification](https://jakarta.ee/specifications/cdi/2.0/cdi-spec-2.0.html#bean_archive), an archive with CDI extension and no `beans.xml` is _not a bean archive_.
That means no bean discovery is performed (i.e. `ContextProducers` class won't be discovered).

I think what you want to have is `beans.xml` with bean discovery mode `annotated` although I don't know every bit of your code :)

Reply:
Added this back, but defaulted to 4.0.